src/Microsoft.VisualStudio.Validation/Requires.cs

```diff
@@ -91,7 +93,7 @@ public static void NotNull([ValidatedNotNull, NotNull]Task value, string paramet
         /// </remarks>
         [DebuggerStepThrough]
         [TargetedPatchingOptOut("Performance critical to inline across NGen image boundaries")]
-        public static void NotNull<T>([ValidatedNotNull, NotNull]Task<T> value, string parameterName)
+        public static void NotNull<T>([ValidatedNotNull, NotNull]Task<T>? value, string? parameterName)
```

Comment:
```suggestion
        public static void NotNull<T>([ValidatedNotNull, NotNull]Task<T?>? value, string? parameterName)
```

Reply:
I can't use `T?` here because there is no `T : class` type constraint. But anyway I'm going the other way and removing the `?` from these `NotNull` methods anyway.

Reply:
My concern here is that passing a `Task<object?>` to this method will give a warning, when it shouldn't.